include/nil/blueprint/input_reader.hpp

```diff
@@ -31,76 +31,121 @@
                 frame(frame), layout_resolver(layout_resolver), memory(memory),
                 assignmnt(assignmnt), public_input_idx(0), private_input_idx(0), constant_idx(0),
                 priv_iter(0), pub_iter(0), has_values(has_values) {}
 
+            InputReader(stack_frame<var> &frame, program_memory<var> &memory, LayoutResolver &layout_resolver) :
+                frame(frame), layout_resolver(layout_resolver), memory(memory), assignmnt(nullptr), public_input_only(true), has_values(true) {
+            }
+
             template<typename InputType>
             var put_into_assignment(InputType &input, bool is_private) {
+                ASSERT(!(public_input_only && is_private));
                 if (is_private) {
-                    assignmnt.private_storage(private_input_idx) = input;
+                    assignmnt->private_storage(private_input_idx) = input;
                     return var(Assignment::private_storage_index, private_input_idx++, false, var::column_type::public_input);
                 } else {
-                    assignmnt.public_input(0, public_input_idx) = input;
+                    if (public_input_only) {
+                        parsed_public_input.push_back(input);
+                        return var();
+                    }
+                    assignmnt->public_input(0, public_input_idx) = input;
                     return var(0, public_input_idx++, false, var::column_type::public_input);
                 }
             }
 
             template<typename InputType>
             var pointer_into_assignment(InputType &ptr) {
-                assignmnt.constant(1, constant_idx) = ptr; // TODO: column index is hardcoded but shouldn't be in the future
+                if (public_input_only) {
+                    return var();
+                }
+                assignmnt->constant(1, constant_idx) = ptr; // TODO: column index is hardcoded but shouldn't be in the future
                 return var(1, constant_idx++, false, var::column_type::constant);
             }
 
-            std::vector<var> process_curve (llvm::EllipticCurveType *curve_type, const boost::json::object &value, bool is_private) {
+            bool check_curve(const signature_node &node, llvm::EllipticCurveKind curve_kind) {
+                if (node.elem != json_elem::CURVE) {
+                    error << "Expected curve argument in the input, got \"" /* << node.elem */ << "\"";
+                    return false;
+                }
+                if (node.children.size() > 1) {
+                    error << "Too much";
+                    return false;
+                }
+                if (node.children.size() == 1) {
+                    auto corresponding_elem = [](llvm::EllipticCurveKind curve_kind) {
+                        switch (curve_kind) {
+                            case llvm::ELLIPTIC_CURVE_PALLAS:
+                                return json_elem::PALLAS;
```

Comment:
implicit conversion from `json_elem` to `bool`

Reply:
Nope, this is the return from lambda function